test/test_common/network_utility.cc

```diff
@@ -0,0 +1,33 @@
+#include "network_utility.h"
+
+#include "common/runtime/runtime_impl.h"
+
+#include <mutex>
+
+namespace Network {
+namespace Test {
+namespace {
+const int kFirstEphemeralPort = 40152;
+const int kLastEphemeralPort = 65535;
+
+int selectFirstPort() {
+  uint64_t range = kLastEphemeralPort - kFirstEphemeralPort + 1;
+  Runtime::RandomGeneratorImpl rng;
```

Comment:
I don't like RNGs in tests (at least unless PRNG and we can seed it and know the seed for repeating). This may introduce flakes in some environments (e.g. when some ports are taken and we occasionally, non-repeatably collide.